hasher-matcher-actioner/hmalib/common/config.py

```diff
@@ -0,0 +1,262 @@
+        result = dynamodb.meta.client.get_item(
+            TableName=_TABLE_NAME,
+            Key={
+                "ConfigType": cls.get_config_type(),
+                "ConfigName": name,
+            },
+        )
+        return _dynamodb_item_to_config(cls, result["Item"])
+
+    @classmethod
+    def get_all(cls: t.Type[TConfig]) -> t.List[TConfig]:
+        assert _TABLE_NAME
+        paginator = dynamodb.meta.client.get_paginator("scan")
+
+        response_iterator = paginator.paginate(TableName=_TABLE_NAME)
+
+        ret = []
+        for page in response_iterator:
+            for item in page["Items"]:
+                ret.append(_dynamodb_item_to_config(item["Item"]))
+        return ret
+
+    @staticmethod
+    def initialize(config_table_name: str) -> None:
+        """
+        Initialize the module with the table name.
+
+        Call this just once (preferably from your main or lambda entry point)
+        """
+        global _TABLE_NAME
+        assert _TABLE_NAME is None
+        _TABLE_NAME = config_table_name
+
+
+# Methods that mutate the config are separate
+# to make them easier to spot in the wild
+
+
+def update_config(config: HMAConfig) -> None:
+    """Update or create a config. No locking or versioning!"""
+    assert _TABLE_NAME
+    # TODO - we should probably sanity check here to make sure all the fields
+    #        are the expected types, because lolpython. Otherwise, it will
+    #        fail to deserialize later
+    dynamodb.meta.client.put_item(
+        TableName=_TABLE_NAME,
+        Item=_config_to_dynamodb_item(config),
+    )
+
+
+def delete_config_by_name(config_type: str, name: str) -> None:
+    """Delete a config by name only"""
+    assert _TABLE_NAME
+    dynamodb.meta.client.delete_item(
+        TableName=_TABLE_NAME,
+        Key={
+            "ConfigType": config_type,
+            "ConfigName": name,
+        },
+    )
+
+
+def delete_config(config: HMAConfig) -> None:
+    """Delete a config"""
+    delete_config_by_name(config.get_config_type(), config.name)
+
+
+def _dynamodb_item_to_config(
+    config_cls: t.Type[TConfig], aws_item: t.Dict[str, t.Any]
+) -> "HMAConfig":
+    """Convert the result of a get_item into a config"""
+    kwargs = {}
+    for field in fields(config_cls):
+        aws_field = aws_item.get(field.name)
+        if aws_field is None:
+            continue  # Hopefully missing b/c default or version difference
+        kwargs[field.name] = _aws_field_to_py(field.type, aws_field)
+    kwargs["name"] = aws_item["ConfigName"]
+    assert aws_item["ConfigType"] == config_cls.get_config_type()
+    return config_cls(**kwargs)
+
+
+def _config_to_dynamodb_item(config) -> t.Dict[str, t.Any]:
+    """
+    Convert a config object into what is what goes into the put_item Item arg
+    """
+    item = {
+        field.name: _py_to_aws_field(field.type, getattr(config, field.name))
+        for field in fields(config)
+    }
+    del item["name"]
```

Comment:
Up above I was going to ask about why "ConfigName" instead of just "name". What about changing the instance variable `name` to `config_name` and that way `name` (something fairly common someone might think to use for something) wouldn't be a reserved word, so to speak? I'm thinking you might have a config as follows:

ConfigType: EmergencyContact
ConfigName: ContactID1
Name: Brian <-- there are perhaps lots of Brians

Reply:
`name` is a reserved variable in DynamoDB, which requires extra boilerplate to handle. Thus the column name is ConfigName, and it's special cased during translation.

Reply:
Sounds good!